assets/javascripts/discourse/components/tc-message.js

```diff
@@ -158,6 +199,170 @@ export default Component.extend({
     });
   },
 
+  @discourseComputed("message.reactions.@each")
+  hasReactions(reactions) {
+    return Object.values(reactions).some((r) => r.count > 0);
+  },
+
+  @action
+  showUsersList(reaction) {
+    let list;
+    let usernames = reaction.users.map((u) => u.username).join(", ");
+    if (reaction.reacted) {
+      if (reaction.count === 1) {
+        list = I18n.t("chat.reactions.only_you", { emoji: reaction.emoji });
+      } else if (reaction.count > 1 && reaction.count < 6) {
+        list = I18n.t("chat.reactions.and_others", {
+          usernames,
+          emoji: reaction.emoji,
+        });
+      } else if (reaction.count >= 6) {
+        list = I18n.t("chat.reactions.you_others_and_more", {
+          usernames,
+          emoji: reaction.emoji,
+          more: reaction.count - 5,
+        });
+      }
+    } else {
+      if (reaction.count > 0 && reaction.count < 6) {
+        list = I18n.t("chat.reactions.only_others", {
+          usernames,
+          emoji: reaction.emoji,
+        });
+      } else if (reaction.count >= 6) {
+        list = I18n.t("chat.reactions.others_and_more", {
+          usernames,
+          emoji: reaction.emoji,
+          more: reaction.count - 5,
+        });
+      }
+    }
+    this.set("reactionLabel", list);
+  },
+
+  @action
+  hideUsersList() {
+    this.set("reactionLabel", null);
+  },
+
+  @action
+  startReaction(btnSelector, position) {
+    this.set("emojiPickerIsActive", true);
+    this.appEvents.trigger(
+      "chat-message:reaction-picker-opened",
+      this.message.id
+    );
+
+    schedule("afterRender", () => {
+      this._repositionEmojiPicker(btnSelector, position);
+    });
+  },
+
+  _repositionEmojiPicker(btnSelector, position) {
```

Comment:
This repositioning of the emoji picker feels a little fragile. Could we use Popper instead? 

Also, not sure how this currently works on mobile, you probably went something full width there.

Reply:
Ignore this too, pulled the repo locally, works quite well.